server/services/invidious.ts

```diff
@@ -76,6 +76,17 @@ export default class InvidiousAdapter extends ServiceAdapter {
 
 	allowedHosts: string[] = [];
 
+	/** When true, accept links from hosts not listed in config and probe them. */
+	private autoDiscoverEnabled = false;
+	/** Probe result cache TTL (ms). */
+	private autoDiscoverTTL = 300000; // 5 minutes
+	/**
+	 * Simple in-memory probe cache: host → { ok, ts }.
+	 * ok=true means the host recently answered a known Invidious endpoint.
+	 */
+	private hostProbeCache = new Map<string, { ok: boolean; ts: number }>();
```

Comment:
We can't have this be an in memory cache. It must be in Redis. OTT can be scaled horizontally, which means multiple instances of the monolith can be run side by side and act as effectively one web app, and this is how the official deployment works.

It will also make the logic a lot more simple because Redis can handle the expiration and key eviction.